Implemented Validator class

Bellkross opened the pull request.

Commit:
Implemented Validator class

PolinaShlepakova commented:
Tester has a lot of very similar blocks of code, which are different in the same places (especially it's noticeable in `void Tester::validatorTests()` and `void Tester::digitSequenceTests()`. Don't you think it would be better to create an array of inputs and change those similar blocks of code to one loop?

Also, I believe this piece of code breaks code conventions:
`Validator(const size_t cardNumberLength=16, const size_t pinLength=4);`

Commit:
convention typo fixed

Commit:
Changed braces convention about classes and functions

Commit:
added std::string constructor in digit sequence, changed digitSequenceTests()

Commit:
Removed char* support from validator, changed validator tests()

Bellkross commented:
> Tester has a lot of very similar blocks of code, which are different in the same places (especially it's noticeable in `void Tester::validatorTests()` and `void Tester::digitSequenceTests()`. Don't you think it would be better to create an array of inputs and change those similar blocks of code to one loop?
I agree, thank you for this remark.
I solved this issue and created loop for this test cases.
> Also, I believe this piece of code breaks code conventions:
> `Validator(const size_t cardNumberLength=16, const size_t pinLength=4);`
Yes, it is true. Fixed it.

In my commits I also removed char* support from validator, changed it to std::string.
And I added std::string support in DigitSequence for adding previous feature.

You can see my changes in last two commits.

Commit:
vector initialization with {} removed
